src/modules/xml/filter_gpstext.c

```diff
@@ -0,0 +1,784 @@
+			"geometry family size weight style fgcolour bgcolour olcolour pad halign valign outline" );
+	mlt_filter_set_in_and_out( text_filter, mlt_filter_get_in( filter ), mlt_filter_get_out( filter ) );
+	return mlt_filter_process( text_filter, frame );
+}
+
+/** Destructor for the filter.
+*/
+static void filter_close (mlt_filter filter)
+{
+	private_data* pdata = (private_data*)filter->child;
+
+	default_priv_data(pdata);
+	free(pdata);
+
+	filter->child = NULL;
+	filter->close = NULL;
+	filter->parent.close = NULL;
+	mlt_service_close( &filter->parent );
+}
+
+/** Constructor for the filter.
+*/
+mlt_filter filter_gpstext_init( mlt_profile profile, mlt_service_type type, const char *id, char *arg )
+{
+	mlt_filter filter = mlt_filter_new();
+	private_data* pdata = (private_data*) calloc(1, sizeof(private_data));
+	default_priv_data(pdata);
+	mlt_filter text_filter = mlt_factory_filter( profile, "qtext", NULL );
+
+	if( !text_filter )
+		text_filter = mlt_factory_filter( profile, "text", NULL );
+
+	if( !text_filter )
+		mlt_log_warning( MLT_FILTER_SERVICE(filter), "Unable to create text filter.\n" );
+
+	if ( filter && text_filter && pdata)
+	{
+		mlt_properties my_properties = MLT_FILTER_PROPERTIES( filter );
+
+		// Register the text filter for reuse/destruction
+		mlt_properties_set_data( my_properties, "_text_filter", text_filter, 0, ( mlt_destructor )mlt_filter_close, NULL );
+
+		// Assign default values
+		mlt_properties_set_string( my_properties, "argument", arg ? arg:
+				"Speed: #gps_speed#km/h\n"
+				"Distance: #gps_dist#m\n"
+				"Altitude: #gps_elev#m\n\n"
+				"GPS time: #gps_datetime_now# UTC\n"
+				"GPS location: #gps_lat#, #gps_lon#"
+				);
+		mlt_properties_set_string( my_properties, "geometry", "10%/10%:80%x80%:100%" );
+		mlt_properties_set_string( my_properties, "family", "Sans" );
+		mlt_properties_set_string( my_properties, "size", "26" );
+		mlt_properties_set_string( my_properties, "weight", "400" );
+		mlt_properties_set_string( my_properties, "style", "normal" );
+		mlt_properties_set_string( my_properties, "fgcolour", "0xffffffff" );
+		mlt_properties_set_string( my_properties, "bgcolour", "0x00000000" );
+		mlt_properties_set_string( my_properties, "olcolour", "0x000000ff" );
+		mlt_properties_set_string( my_properties, "pad", "5" );
+		mlt_properties_set_string( my_properties, "halign", "left" );
+		mlt_properties_set_string( my_properties, "valign", "bottom" );
+		mlt_properties_set_string( my_properties, "outline", "0" );
+		mlt_properties_set_int( my_properties, "_filter_private", 1 );
+
+		mlt_properties_set_int( my_properties, "major_offset", 0);
+		mlt_properties_set_int( my_properties, "minor_offset", 0);
+		mlt_properties_set_int( my_properties, "smoothing_value", 5);
+		mlt_properties_set_int( my_properties, "videofile_timezone_seconds", 0);
+		mlt_properties_set_int( my_properties, "speed_multiplier", 1);
+		mlt_properties_set_int( my_properties, "updates_per_second", 1);
+
```

Comment:
Question: can I get a reference to a mlt_producer in filter_gpstext_init here? I am thinking to set video_start here so it is available since the begining and also to not update it every frame due to filter_process(). 
I need it to call mlt_producer_get_creation_time(producer), but the only way I get producer is tied to a frame: mlt_frame_get_original_producer( frame )

Reply:
That is not possible. A filter does not know what producer is going to generate the frames that it will eventually process.

Reply:
Ok, I thought the filter init is called after the producer has info about the video file but it's probably more complicated than this.